src/scripts/main.js

```diff
@@ -1,3 +1,221 @@
 'use strict';
 
-// write code here
+//Sorting by header of column: 
+
+const body = document.querySelector('body');
+const table = document.querySelector('table');
+const headers = table.querySelectorAll('th');
+const tBody = table.querySelector('tbody');
+const cells = tBody.querySelectorAll('td');
+const td = document.createElement('td');
+let rows = [...tBody.querySelectorAll('tr')];
+const sortOrder = Array(headers.length).fill(true);
+
+function updateRows() {
+  rows = [...tBody.querySelectorAll('tr')];
+}
+
+headers.forEach((header, index) => {
+  header.addEventListener('click', () => {
+    setSortTable(index);
+  });
+});
+
+const setSortTable = (columnIndex) => {
+  updateRows();
+
+  const sortedRows = rows.sort((a, b) => {
+    const aValue = a.cells[columnIndex].textContent.trim();
+    const bValue = b.cells[columnIndex].textContent.trim();
+
+    if (+aValue) {
+      return sortOrder[columnIndex] ? +aValue - bValue : bValue - aValue;
```

Comment:
The type coercion in the condition `if (+aValue)` can lead to unexpected results if `aValue` is not a number. Consider using `isNaN()` to check if the value is a number before performing arithmetic operations.